truffle/src/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/CompilationState.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.truffle.api.instrumentation;
+
+/*
+ * Thermometer TODO: I'd rather not have this whole class, and instead have an
+ * instrument event listener for updates.
+ */
+public interface CompilationState {
+
+    int getQueued();
+
+    int getRunning();
+
+    int getFinished();
+
+    int getFailed();
+
+    int getDequeued();
+
+    int getDeoptimizations();
```

Comment:
Is there a reason you are capturing deoptimizations and not invalidations here?

Reply:
I simplified both deoptimisations and invalidations into a general concept of 'code that had to be thrown away' and called that deoptimisations.

Reply:
if code is thrown away it is always invalidated and made non reentrant. Deoptimization can happen for other reasons too (e.g. CompilerDirectives.transferToInterpreter), but does not imply invalidation. Because of that I think that terminology is misleading.